sdk/endpoint/auth.go

```diff
@@ -0,0 +1,55 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/open-policy-agent/opa/rego"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+)
+
+type AuthEndpoint struct {
+	policy rego.PreparedEvalQuery
+}
+
+func (auth *AuthEndpoint) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+	rs, err := auth.policy.Eval(ctx, rego.EvalInput(request))
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	for _, r := range rs {
+		for _, e := range r.Expressions {
+			t, ok := e.Value.(bool)
+			if !ok {
+				return nil, status.Error(codes.Internal, "non boolean expression")
+			}
+
+			if !t {
+				return nil, status.Error(codes.PermissionDenied, "no sufficient privileges to call Request")
```

Comment:
Is it possible to get any richer data about why permission Denied to help the user?

Reply:
Yes but I think it depends on provided policy? Now `AuthzEndpoint` is quite generic. But I guess I can add some information about Spiffe Id